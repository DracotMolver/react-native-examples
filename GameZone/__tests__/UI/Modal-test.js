import React from 'react';
import { render, cleanup } from 'react-native-testing-library';
// mock
// component
import ModalHOC from '../../app/components/UI/Modal';

afterEach(cleanup);

describe('Testing UI - Modal component', () => {
  const FakeComponent = ({ title }) => {
    const { Modal } = ModalHOC();

    return <Modal title={title} />;
  };

  it('Should render right', () => {
    const rendered = render(<FakeComponent title="test" />).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it('Should start closed (false)', () => {
    const { getByTestId } = render(<FakeComponent title="test" />);

    const modal = getByTestId('modal-native');

    expect(modal.props.visible).toBeFalsy();
  });

  it('Should hast the right text', () => {
    const { getByTestId } = render(<FakeComponent title="test" />);

    const text = getByTestId('modal-text');

    expect(text.props.children).toBe('test');
  });
});
