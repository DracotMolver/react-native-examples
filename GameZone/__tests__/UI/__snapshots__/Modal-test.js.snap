// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UI - Modal component Should render right 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  testID="modal-native"
  transparent={false}
  visible={false}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 16,
        }
      }
    >
      <Text
        testID="modal-text"
      >
        test
      </Text>
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "color": "blue",
              "fontSize": 32,
            },
            Object {
              "borderColor": "#f2f2f2",
              "borderRadius": 3,
              "borderWidth": 1,
              "marginLeft": 16,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="modal-icon"
      >
        Óóç
      </Text>
    </View>
  </RCTSafeAreaView>
</Modal>
`;
